File: python/templates/bfs_graph.py
from collections import deque

# Define the graph as an adjacency list
graph = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': ['F'],
    'D': [],
    'E': ['F'],
    'F': []
}

# Function to perform Breadth-First Search (BFS)
def bfs(start):
    visited = {}
    queue = deque()

    queue.append(start)
    visited[start] = True

    while queue:
        node = queue.popleft()
        print(node, end=' ')

        for neighbor in graph[node]:
            if neighbor not in visited:
                queue.append(neighbor)
                visited[neighbor] = True

# Example usage:
if __name__ == "__main__":
    print("Breadth-First Search (BFS) Traversal:")
    bfs('A')
    print()

File: javascript/templates/bfs_graph.js
// Define the graph as an adjacency list
const graph = {
    'A': ['B', 'C'],
    'B': ['D', 'E'],
    'C': ['F'],
    'D': [],
    'E': ['F'],
    'F': []
};

// Function to perform Breadth-First Search (BFS)
function bfs(start) {
    const visited = {};
    const queue = [];

    queue.push(start);
    visited[start] = true;

    while (queue.length > 0) {
        const node = queue.shift();
        process.stdout.write(node + ' ');

        for (const neighbor of graph[node]) {
            if (!visited[neighbor]) {
                queue.push(neighbor);
                visited[neighbor] = true;
            }
        }
    }
}

// Example usage:
console.log('Breadth-First Search (BFS) Traversal:');
bfs('A');
console.log();